commands/command_migrate_export.go

```diff
@@ -104,9 +116,25 @@ func migrateExportCommand(cmd *cobra.Command, args []string) {
 	// If we have a valid remote, pre-download all objects using the Transfer Queue
 	if remoteURL := getAPIClient().Endpoints.RemoteEndpoint("download", cfg.Remote()).Url; remoteURL != "" {
 		q := newDownloadQueue(getTransferManifestOperationRemote("Download", cfg.Remote()), cfg.Remote())
-		if err := rewriter.ScanForPointers(q, opts, gitfilter); err != nil {
-			ExitWithError(err)
-		}
+		gs := lfs.NewGitScanner(func(p *lfs.WrappedPointer, err error) {
+			if err != nil {
+				return
+			}
+
+			if !filter.Allows(p.Name) {
+				return
+			}
+
+			downloadPath, err := gitfilter.ObjectPath(p.Oid)
+			if err != nil {
+				return
+			}
+
+			if _, err := os.Stat(downloadPath); os.IsNotExist(err) {
```

Comment:
Nice. I wonder if we could add this to the `Add()` function on the transfer queue itself. I can imagine that other callers would find this useful.

Reply:
I just tried this, but unfortunately it led to a bunch of inexplicable failures about objects not being found by the transfer queue when running the `export` tests. I verified that the conditional I added to the function is indeed correct so I'm not sure what's going wrong - it's definitely trying to download the files since they don't exist on disk. This problem bled over into other tests that leverage the transfer queue as well, such as `fetch`.

I can try to do some additional debugging but I'm inclined to leave this as-is for now. This is here to prevent download of an object that exists on disk but isn't present on the remote (say, if a recently-migrated pointer hasn't been pushed up to the remote repository yet). The transfer queue is leveraged by lots of code in Git LFS, and it seems like re-downloading an object file won't necessarily be wrong in all cases

Reply:
> I can try to do some additional debugging but I'm inclined to leave this as-is for now. 

That's fine, let's mark this as #leftoverbits, and touch this in a future pull request.